osu.Game/Database/RealmObjectExtensions.cs

```diff
@@ -291,7 +292,21 @@ public static IDisposable QueryAsyncWithNotifications<T>(this IRealmCollection<T
             if (!RealmAccess.CurrentThreadSubscriptionsAllowed)
                 throw new InvalidOperationException($"Make sure to call {nameof(RealmAccess)}.{nameof(RealmAccess.RegisterForNotifications)}");
 
-            return collection.SubscribeForNotifications(callback);
+            bool initial = true;
+            return collection.SubscribeForNotifications(((sender, changes) =>
+            {
+                if (initial)
+                {
+                    initial = false;
+
+                    // Realm might coalesce the initial callback, meaning we never receive a `ChangeSet` of `null` marking the first callback.
+                    // Let's decouple it for simplicity in handling.
+                    if (changes != null)
+                        callback(sender, null);
+                }
+
+                callback(sender, changes);
```

Comment:
If `sender` contains the items _after_ the modifications are applied, then shouldn't we _not_ send the callback if `initial == true`? That is, currently:

To further expound, when you send `(sender, null)`, the callsite will use those items to populate the initial list:

https://github.com/ppy/osu/blob/c100d1ab65b23401e5e2533636f4b32d3e4f698c/osu.Game/Screens/Select/BeatmapCarousel.cs#L279-L282

But then we send the `(sender, changes)` callback which could be used to e.g. delete from that "initial list" from above:

https://github.com/ppy/osu/blob/c100d1ab65b23401e5e2533636f4b32d3e4f698c/osu.Game/Screens/Select/BeatmapCarousel.cs#L290-L294

But the deleted indices are actually irrelevant here because we did not previously have an initial list.

It depends on what `sender` truly is in the case when the changes have been coalesced - is it the initial list _before_ any modifications, or is it the list _after_ all coalesced modifications.

Reply:
Yeah just noticed this very thing in further testing.